src/Transaction/components/PaymentAccountSelectionDialog.tsx

```diff
@@ -0,0 +1,249 @@
+import React from "react"
+import { Trans, useTranslation } from "react-i18next"
+import { Asset } from "stellar-sdk"
+import { PayStellarUri } from "@stellarguard/stellar-uri"
+import Box from "@material-ui/core/Box"
+import Dialog from "@material-ui/core/Dialog"
+import Grid from "@material-ui/core/Grid"
+import makeStyles from "@material-ui/core/styles/makeStyles"
+import Typography from "@material-ui/core/Typography"
+import CancelIcon from "@material-ui/icons/Cancel"
+import SelectIcon from "@material-ui/icons/Check"
+import WarningIcon from "@material-ui/icons/Warning"
+import { Account, AccountsContext } from "~App/contexts/accounts"
+import { FullscreenDialogTransition, warningColor, breakpoints } from "~App/theme"
+import AccountSelectionList from "~Account/components/AccountSelectionList"
+import AssetLogo from "~Assets/components/AssetLogo"
+import { ActionButton, DialogActionsBox } from "~Generic/components/DialogActions"
+import { CopyableAddress } from "~Generic/components/PublicKey"
+import MainTitle from "~Generic/components/MainTitle"
+import ViewLoading from "~Generic/components/ViewLoading"
+import { balancelineToAsset } from "~Generic/lib/stellar"
+import { useLiveAccountDataSet } from "~Generic/hooks/stellar-subscriptions"
+import { useIsMobile } from "~Generic/hooks/userinterface"
+import DialogBody from "~Layout/components/DialogBody"
+import PaymentDialog from "~Payment/components/PaymentDialog"
+
+const useStyles = makeStyles(() => ({
+  assetContainer: {
+    alignSelf: "center",
+    display: "flex",
+    margin: "0px 8px"
+  },
+  assetLogo: {
+    width: 28,
+    height: 28,
+    margin: "0px 4px"
+  },
+  root: {
+    display: "flex",
+    flexDirection: "column",
+    padding: "12px 0 0"
+  },
+  row: {
+    lineHeight: 1.2
+  },
+  keyTypography: {
+    alignSelf: "center",
+    textAlign: "right"
+  },
+  valueTypography: {
+    textAlign: "left"
+  },
+  uriContainer: {
+    paddingTop: 32,
+    paddingBottom: 32
+  },
+  warningContainer: {
+    alignItems: "center",
+    alignSelf: "center",
+    background: warningColor,
+    display: "flex",
+    justifyContent: "center",
+    padding: "6px 16px",
+    width: "fit-content",
+
+    [breakpoints.up(600)]: {
+      width: "100%"
+    }
+  }
+}))
+
+interface PaymentAccountSelectionDialogProps {
+  payStellarUri: PayStellarUri
+  onClose: () => void
+}
+
+function PaymentAccountSelectionDialog(props: PaymentAccountSelectionDialogProps) {
```

Comment:
One thing that we can either refactor right now or we need to do it as soon as we start working on `tx` type requests:

It should probably rather be called `TransactionRequestReviewDialog` or similar and we should pass that SEP-7 URI content rendered as a grid as `children` prop, so we can re-use this component for the `tx` type URIs.

Reply:
I suggest we postpone it until we do the `tx` type requests because it might be easier to restructure once we know how we handle/show `tx` requests.